src/plot_api/accessibility.js

```diff
@@ -0,0 +1,71 @@
+"use strict";
+import c2mChart from "chart2music";
+
+export function enable (gd) {
+
+    const c2mData = {};
+    const labels = [];
+    
+    const fullData = gd._fullData;
+
+    for(var i = 0; i < fullData.length; i++) {
+        var trace = fullData[i] ?? {};
+        var {type, x = [], y = [], name = i, text = []} = trace;
+        if(type === 'scatter') {
+            var traceData = [];
+            if ('y' in trace) {
+                for(var p = 0; p < y.length; p++) {
+                    traceData.push(
+                        {
+                            x: x.length > 0 ? x[p] : p,
+                            y: y[p],
+                            label: text[p] ?? p
+                        })
+                }
+                c2mData[name] = traceData;
+                labels.push(name);
+            }
+        }
+        else {
+            // 'Accessibility not implemented for trace type: ' + trace.type
+            return;
+        }
+    }
+    
+    var closed_captions = document.createElement('div');
+    closed_captions.id = 'cc';
+    closed_captions.className = 'closed_captions';
+    gd.appendChild(closed_captions);
+
+    const {
+        title: {text: title_text = ''} = {},
+        xaxis: {title: {text: xaxis_text = ''} = {}} = {},
+        yaxis: {title: {text: yaxis_text = ''} = {}} = {},
+    } = gd._fullLayout;
+    
+    c2mChart({
+        title: title_text,
+        type: "line",
+        axes: {
+            x: {
+            label: xaxis_text
+            },
+            y: {
+            label: yaxis_text
+            }
+        },
+        element: gd,
+        cc: closed_captions,
+        data: c2mData,
+        options: {
+            onFocusCallback: ({slice, index}) => {
+            Plotly.Fx.hover(gd, [{
+                curveNumber: labels.indexOf(slice), 
+                pointNumber: index
+            }])
+            },
+            ...gd._context.chart2musicOptions
+        },
+        info: gd._context.chart2musicInfo
```

Comment:
Maybe we can combine these into one and [`extendDeep`](https://github.com/plotly/plotly.js/blob/149c895db2acf3f497a84d62adc276c24d37989b/src/lib/extend.js#L24) the config object?

Also I was discussing this PR with @archmoj and he had a good idea, instead of
```js
config: {chart2music: true, chart2musicConfig: {...}}
```
how about we make an `accessibility` object like:
```js
config: {
    accessibility: {
        library: 'chart2music', // easier to extend, but for now chart2music would be the only option, and the default if you make an accessibility object.
        enabled: true, // optional but would allow disabling if necessary
        options: {...}
    }
}
```

Reply:
I like that. That was the idea behind the file naming, makes sense to have the same structure in the config

Reply:
These changes have been made, I didn't use extendDeep (just because I wasn't sure on usage) but options and info are now combined